libheif/uncompressed_image.cc

```diff
@@ -25,38 +25,55 @@
 #include <map>
 #include <iostream>
 
+#include "libheif/heif.h"
 #include "uncompressed_image.h"
 
 
-enum heif_component_type
+enum uncompressed_component_type
 {
-  heif_component_type_monochrome = 0,
-  heif_component_type_Y = 1,
-  heif_component_type_Cb = 2,
-  heif_component_type_Cr = 3,
-  heif_component_type_red = 4,
-  heif_component_type_green = 5,
-  heif_component_type_blue = 6,
-  heif_component_type_alpha = 7,
-  heif_component_type_depth = 8,
-  heif_component_type_disparity = 9,
-  heif_component_type_palette = 10,
-  heif_component_type_filter_array = 11,
-  heif_component_type_padded = 12
+  component_type_monochrome = 0,
+  component_type_Y = 1,
+  component_type_Cb = 2,
+  component_type_Cr = 3,
+  component_type_red = 4,
+  component_type_green = 5,
+  component_type_blue = 6,
+  component_type_alpha = 7,
+  component_type_depth = 8,
+  component_type_disparity = 9,
+  component_type_palette = 10,
+  component_type_filter_array = 11,
+  component_type_padded = 12
 };
 
 
-enum heif_uncompressed_interleave_type
+enum uncompressed_component_format
 {
-  heif_uncompressed_interleave_type_component = 0,
-  heif_uncompressed_interleave_type_pixel = 1,
-  heif_uncompressed_interleave_type_mixed = 2,
-  heif_uncompressed_interleave_type_row = 3,
-  heif_uncompressed_interleave_type_tile_component = 4,
-  heif_uncompressed_interleave_type_multi_y = 5
+  component_format_unsigned = 0,
+  component_format_float = 1,
+  component_format_complex = 2,
 };
 
 
+enum uncompressed_sampling_type
```

Comment:
I think `heif_chroma` does more than just sampling type (e.g. interleave type and bit depth). That is useful in that you only get combinations that work, but does represent a bit of a mismatch when parsing and when using these values as constants for write-out.

Reply:
Yes, I agree. Let's keep the uncompressed chroma type separate. Just like the 'heif_components' are separate from 'heif_channel'.
Maybe we could even make this more clear by naming the uncompressed-specific types with a `heif_unc_` prefix. But this can also be done later when this is exposed in a special API.